File: Java/백준/Gold/10775. 공항/공항.java
import java.io.*;
import java.util.*;

public class Main {
    static int[] parent;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int G = Integer.parseInt(br.readLine());
        int P = Integer.parseInt(br.readLine());

        parent = new int[G + 1];
        for (int i = 0; i <= G; i++) {
            parent[i] = i;
        }

        int result = 0;
        for (int i = 0; i < P; i++) {
            int gi = Integer.parseInt(br.readLine());
            int root = find(gi);
            if (root == 0) break;
            union(root, root - 1);
            result++;
        }

        System.out.println(result);
    }

    static int find(int x) {
        if (parent[x] != x) {
            parent[x] = find(parent[x]);
        }
        return parent[x];
    }

    static void union(int a, int b) {
        a = find(a);
        b = find(b);
        parent[a] = b;
    }
}

File: C11/백준/Gold/10775. 공항/공항.c
#include <stdio.h>

int parent[100001];

int find(int x) {
    if (parent[x] != x)
        parent[x] = find(parent[x]);
    return parent[x];
}

void union_set(int a, int b) {
    a = find(a);
    b = find(b);
    parent[a] = b;
}

int main() {
    int G, P, gi, result = 0;
    scanf("%d", &G);
    scanf("%d", &P);

    for (int i = 0; i <= G; i++) {
        parent[i] = i;
    }

    for (int i = 0; i < P; i++) {
        scanf("%d", &gi);
        int root = find(gi);
        if (root == 0)
            break;
        union_set(root, root - 1);
        result++;
    }

    printf("%d\n", result);
    return 0;
}